### ISSUE TITLE
False positive for Generic.PHP.LowerCaseKeyword.Found when having constant with key

### ISSUE THREAD

**Issue by dakorpar** [squizlabs/PHP_CodeSniffer#3280]
**Describe the bug**

**Code sample**
```php
class SomeClass {
    public const DEFAULT = 'whatever';
}
...
		$arr = [
			SomeClass::DEFAULT => 1,
		]
SomeClass::DEFAULT //trigers error PHP keywords must be lowercase; expected "default" but found "DEFAULT"
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
ERROR | [x] PHP keywords must be lowercase; expected "default" but found "DEFAULT"
    |       |     (Generic.PHP.LowerCaseKeyword.Found)
ERROR | [ ] No key specified for array entry; first entry specifies key
    |       |     (Squiz.Arrays.ArrayDeclaration.NoKeySpecified)

```

**Expected behavior**
No error triggered
**Versions (please complete the following information):**
 - OS: Ubuntu 20.4
 - PHP: [8.0]
 - PHPCS: [* dev-master, * 3.x-dev]
 - Standard: [e.g., PSR2, PSR12, Squiz]

**Comment by jrfnl** [comment]
@dakorpar Nice! You actually found a tokenizer bug. I'll look into the tokenizer issue first and will look at the sniff side of things after that (if still needed once the tokenizer issue has been fixed).

**Comment by jrfnl** [comment]
@dakorpar This should be fixed now via PR #3282. Testing appreciated.

**Comment by jrfnl** [comment]
Oh and just for completeness: as I suspected, fixing the tokenizer issue automatically fixed the sniff issue.

**Comment by dakorpar** [comment]
@jrfnl I'm trying to get this to test but I can't install it, allways some conflicts with other packages...
I tried with

`		"squizlabs/php_codesniffer": "dev-feature/3280-generic-lowercasekeyword-bug-fix as dev-master",`

composer installs everything with

`		"squizlabs/php_codesniffer": "dev-master",`

but with this I can't get all dependencies installed...
any idea?

**Comment by jrfnl** [comment]
@dakorpar The dependencies of PHPCS haven't changed since 3.0.0, so I'm not sure what conflict you are seeing.

Do you have the `repositories` key set in your Composer file though ? The branch is in a fork, not in this repo.

```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/jrfnl/PHP_CodeSniffer"
        }
    ],
    "require": {
        "squizlabs/php-php_codesniffer": "dev-feature/3280-generic-lowercasekeyword-bug-fix"
    }
}
```

You could use `as 3.6.0` as the alias as it will (should) be merged into what will be released as that version.

Only other thing I can think of off hand is to check if the `minimum-stability` key is set to `stable` in your `composer.json`. If so, you could try setting it to `dev` temporarily to test the branch.

**Comment by gsherwood** [comment]
Thanks for reporting this. I couldn't replicate in the latest stable release so it's not in the changelog, but please let me know if I've missed a test case for it.

**Comment by dakorpar** [comment]
@jrfnl Sorry for being slow, to much things,
I've had url set to your fork, it's not the problem in codesniffer, but I have few other packages that require phpcs 3.6 and  I was using dev-master for all of them and everything installs well, but as soon as I changed version to your branch other dependencies couldn't install because their requirements were different that's why I added dev-master alias but that also didn't help so I was stuck. Will check now when it's in master :)

**Comment by dakorpar** [comment]
@gsherwood  @jrfnl Everything works now in dev-master, thanks!

**Comment by jrfnl** [comment]
@dakorpar Thanks for testing and confirming the fix works!

@gsherwood Correct - this was an issue only in `dev-master` related to the `match - default` token backfill and not reproducible in any released version.
